src/dstack/_internal/core/backends/aws/resources.py

```diff
@@ -6,22 +6,45 @@
 
 import dstack.version as version
 from dstack._internal.core.errors import BackendError, ComputeError, ComputeResourceNotFoundError
+from dstack._internal.core.models.backends.aws import AWSOSImageConfig
+from dstack._internal.utils.logging import get_logger
 
+logger = get_logger(__name__)
+
+DSTACK_ACCOUNT_ID = "142421590066"
 
-def get_image_id(ec2_client: botocore.client.BaseClient, cuda: bool) -> str:
-    image_name = (
-        f"dstack-{version.base_image}" if not cuda else f"dstack-cuda-{version.base_image}"
-    )
 
-    response = ec2_client.describe_images(Filters=[{"Name": "name", "Values": [image_name]}])
+def get_image_id_and_username(
+    ec2_client: botocore.client.BaseClient,
+    cuda: bool,
+    image_config: Optional[AWSOSImageConfig] = None,
+) -> tuple[str, str]:
+    if image_config is not None:
+        image = image_config.nvidia if cuda else image_config.cpu
+        if image is None:
+            logger.warning("%s image not configured", "nvidia" if cuda else "cpu")
+            raise ComputeResourceNotFoundError()
```

Comment:
(nit) Btw, alternatively we could filter out the offers that don't have an image configured. This way, we won't waste provisioning attempts and will show more realistic run plans. That would especially be useful for Azure, e.g. to filter out A10 offers if the GRID image is not configured.

Yet the current implementation provides more insight into why the offers aren't working, so I'm not sure which alternative is better and we can stick with the current one.

Reply:
As for now, I would stick to the current implementation, as I assume that the setup where custom AMIs are used, but not all (currently, all = only two) kinds are configured, is rare. In the future, if there are more kinds of images, we could revisit the logic.